modules/core/src/lib/layer.ts

```diff
@@ -1079,6 +1079,15 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
         const {modelMatrix} = this.props;
         this.setModuleParameters(moduleParameters);
         const {
+          // shadow
```

Comment:
Concern: 
- That is a lot of individual imports, kind of pollutes to have such a big surface area from a leaf functionality into a core file. 
- Same for terrain I suppose though I didn't catch it then.
- Is there a way things can be made more decoupled or pluggable. Some function to get the props, instead of requiring layer class to be aware of them all?

Reply:
Agree, I have a [TODO](https://github.com/visgl/deck.gl/pull/9054/files/916f3409a3be1089ff916920fbd3e826b1801605#diff-6865938bd9e33227cb4501cefc20abe93abb4098905de00a0653cd1a80750544R1122) to review it once the UBO migration is complete. I chose to do it here as this is where `setModuleParameters` is being invoked in the same way.

Some of the other modules are already self-contained, like the `MaskExtension`, so I expect we will be able to tidy this up. First though I would like to remove all the `updateModuleSettings` calls before potentially changing the data flow in the code.

I have added this point to the Audit Log as suggested